apps/halide/blur/main.cpp

```diff
@@ -96,40 +96,23 @@ bool write_png_file(
 }
 
 typedef void (*blurtype)(
-    void *ctxt, int_fast32_t n, int_fast32_t m, uint8_t *g, const uint8_t *inp);
+    void *ctxt, int_fast32_t n, int_fast32_t m, uint16_t *g, const uint16_t *inp);
 
 int exec_parrot(blurtype func, std::string output_name, int width, int height,
-    uint8_t *parrot) {
-  uint8_t *parrot_blurred_0, *parrot_blurred_1;
-  parrot_blurred_0 =
-      (uint8_t *)malloc(sizeof(uint8_t) * (width + 4) * (height + 4));
-  parrot_blurred_1 =
-      (uint8_t *)malloc(sizeof(uint8_t) * (width + 4) * (height + 4));
+    uint16_t *parrot) {
+  uint16_t *parrot_write = (uint16_t *)malloc(sizeof(uint16_t) * width * height);
```

Comment:
You have a memory leak here. You should add a corresponding free or wrap it with a smart pointer. Another option is to use a std::vector and provide the size of the buffer to the constructor. Then, you can extract the raw pointer with std::vector::data()